sources/packages/api/src/route-controllers/confirmation-of-enrollment/confirmation-of-enrollment.controller.ts

```diff
@@ -51,21 +54,29 @@ export class ConfirmationOfEnrollmentController {
    * @returns student application list of an institution location
    */
   @HasLocationAccess("locationId")
-  @Get(":locationId/confirmation-of-enrollment")
+  @Get(
+    ":locationId/confirmation-of-enrollment/enrollmentPeriod/:enrollmentPeriod",
+  )
   async getCOESummary(
     @Param("locationId") locationId: number,
+    @Param("enrollmentPeriod") enrollmentPeriod: string,
   ): Promise<COESummaryDTO[]> {
-    const applications = await this.applicationService.getCOEApplications(
-      locationId,
-    );
-    return applications.map((eachApplication: Application) => {
+    const disbursementSchedules =
+      await this.disbursementScheduleService.getCOEByLocation(
+        locationId,
+        enrollmentPeriod === EnrollmentPeriod.Upcoming,
+      );
+    return disbursementSchedules.map((disbursement: DisbursementSchedule) => {
       return {
-        applicationNumber: eachApplication.applicationNumber,
-        applicationId: eachApplication.id,
-        studyStartPeriod: eachApplication.offering?.studyStartDate ?? "",
-        studyEndPeriod: eachApplication.offering?.studyEndDate ?? "",
-        coeStatus: eachApplication.coeStatus,
-        fullName: getUserFullName(eachApplication.student.user),
+        applicationNumber: disbursement.application.applicationNumber,
+        applicationId: disbursement.application.id,
+        studyStartPeriod:
+          disbursement.application.offering?.studyStartDate ?? "",
+        studyEndPeriod: disbursement.application.offering?.studyEndDate ?? "",
+        coeStatus: this.getCOEStatus(disbursement.isCOEApproved),
```

Comment:
Are we not supposed to be using COE status instead of a Boolean here?

Reply:
Here I am parsing the boolean from db column to COEStatus only.

Reply:
isCOEApproved was changed to an enum and this code was adapted as well.